services/wallet/thirdparty/coingecko/client.go

```diff
@@ -0,0 +1,298 @@
+package coingecko
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+	"github.com/status-im/status-go/services/wallet/thirdparty/utils"
+)
+
+var coinGeckoMapping = map[string]string{
+	"STT":  "status",
+	"SNT":  "status",
+	"ETH":  "ethereum",
+	"AST":  "airswap",
+	"AMB":  "",
+	"ABT":  "arcblock",
+	"ATM":  "",
+	"BNB":  "binancecoin",
+	"BLT":  "bloom",
+	"CDT":  "",
+	"COMP": "compound-coin",
+	"EDG":  "edgeless",
+	"ELF":  "",
+	"ENG":  "enigma",
+	"EOS":  "eos",
+	"GEN":  "daostack",
+	"MANA": "decentraland-wormhole",
+	"LEND": "ethlend",
+	"LRC":  "loopring",
+	"MET":  "metronome",
+	"POLY": "polymath",
+	"PPT":  "populous",
+	"SAN":  "santiment-network-token",
+	"DNT":  "district0x",
+	"SPN":  "sapien",
+	"USDS": "stableusd",
+	"STX":  "stox",
+	"SUB":  "substratum",
+	"PAY":  "tenx",
+	"GRT":  "the-graph",
+	"TNT":  "tierion",
+	"TRX":  "tron",
+	"TGT":  "",
+	"RARE": "superrare",
+	"UNI":  "uniswap",
+	"USDC": "usd-coin",
+	"USDP": "paxos-standard",
+	"VRS":  "",
+	"TIME": "",
+}
+
+const baseURL = "https://api.coingecko.com/api/v3/"
+
+type HistoricalPriceContainer struct {
+	Prices [][]float64 `json:"prices"`
+}
+type GeckoMarketValues struct {
+	ID                                string  `json:"id"`
+	Symbol                            string  `json:"symbol"`
+	Name                              string  `json:"name"`
+	MarketCap                         float64 `json:"market_cap"`
+	High24h                           float64 `json:"high_24h"`
+	Low24h                            float64 `json:"low_24h"`
+	PriceChange24h                    float64 `json:"price_change_24h"`
+	PriceChangePercentage24h          float64 `json:"price_change_percentage_24h"`
+	PriceChangePercentage1hInCurrency float64 `json:"price_change_percentage_1h_in_currency"`
+}
+
+type GeckoToken struct {
+	ID     string `json:"id"`
+	Symbol string `json:"symbol"`
+	Name   string `json:"name"`
+}
+
+type Client struct {
+	client *http.Client
+	tokens map[string]GeckoToken
+}
+
+func NewClient() *Client {
+	return &Client{client: &http.Client{Timeout: time.Minute}, tokens: make(map[string]GeckoToken)}
+}
+
+func (c *Client) DoQuery(url string) (*http.Response, error) {
+	resp, err := c.client.Get(url)
+
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (c *Client) getTokens() (map[string]GeckoToken, error) {
+	if len(c.tokens) > 0 {
+		return c.tokens, nil
+	}
+
+	url := fmt.Sprintf("%scoins/list", baseURL)
+	resp, err := c.DoQuery(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
```

Comment:
Should we handle error codes here? https://support.coingecko.com/hc/en-us/articles/6472757474457-How-can-I-differentiate-between-the-status-codes-I-am-receiving-and-what-do-they-mean-

Reply:
I didn't thought about handling error as it will be part of the circuit. If error => go to next.

If we manage error we will have to unify errors across all the "provider". It would be nice to do but I don't yet know what to do with the errors :thinking: